### PR TITLE
Add a getzone script into toolbox

### PR DESCRIPTION
**Pull request by Winael**

### CODE REVIEW

File: pkgs/getzones/getzones.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    infra_zone_name=$1
+    shift
+    ;;
+  esac
+done
+
+refresh_zones
+
+if zone_contains "$infra_zone_name" "$INFRA_ZONES_NAMES"; then
+  log-debbug "Found caascad infra zone name ${infra_zone_name}"
```
Comment (Benjile) [inline pkgs/getzones/getzones.sh:142]: log_debug i think

File: pkgs/getzones/getzones.sh [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  infra-stg
+  infra-prd
+)
+
+log() {
+  echo -e "\x1B[32m--- $*\x1B[0m" >&2
+}
+
+log_debug() {
+  if [ $KSWITCH_DEBUG -eq 1 ]; then
```
Comment (Benjile) [inline pkgs/getzones/getzones.sh:21]: you should rename this variable

File: pkgs/getzones/getzones.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+# shellcheck disable=SC2086
+
+set -eu
+set -o pipefail
+
+CAASCAD_ZONES_URL=${CAASCAD_ZONES_URL:-https://git.corp.cloudwatt.com/caascad/caascad-zones/raw/master/zones.json}
+CONFIG_DIR="$HOME/.config/kswitch"
+CAASCAD_ZONES_FILE="$CONFIG_DIR/zones.json"
+KSWITCH_DEBUG=${KSWITCH_DEBUG:-0}
+INFRA_ZONES_NAMES=(
```
Comment (Benjile) [inline pkgs/getzones/getzones.sh:11]: can be built from caascad zones, it should not be static

File: pkgs/getzones/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "getzones";
+  version = "0.0.1";
+
+  buildInputs = [ makeWrapper ];
+  passAsFile = [ "buildCommand" ];
+  buildCommand = ''
+    mkdir -p $out/bin $out/share/bash-completion/completions
+    cp ${./getzones.sh} $out/bin/getzones
```
Comment (eonpatapon) [inline pkgs/getzones/default.nix:17]: Same here, we need to find a good name

File: pkgs/getzones/getzones.sh [hunk 5]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2086
+
+set -eu
+set -o pipefail
+
+CAASCAD_ZONES_URL=${CAASCAD_ZONES_URL:-https://git.corp.cloudwatt.com/caascad/caascad-zones/raw/master/zones.json}
+CONFIG_DIR="$HOME/.config/kswitch"
```
Comment (eonpatapon) [inline pkgs/getzones/getzones.sh:8]: Change this path if you need it

File: pkgs/getzones/getzones.sh [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+usage() {
+  cat <<EOF
+Usage:
+  getzones PARENT_ZONE_NAME     Get a zones list of OCB zones and client zones attached to an INFRA_ZONE_NAME
+  getzones -h, --help           This help
+  getzones -v, --version        Current getzones version
+
+getzones automatically give a list of OCB and Client zones attached to an INFRA_ZONE_NAME.
```
Comment (eonpatapon) [inline pkgs/getzones/getzones.sh:67]: To be useful I think it should handle the most common case: find child zones of some parent zone

**Review by eonpatapon — CHANGES_REQUESTED** [review]

File: pkgs/getzones/getzones.sh (hunk 6 above)
Comment (Winael) [reply pkgs/getzones/getzones.sh:67]: It is not the same, but with a better explanation ?

**Pull request closed without merging**